Edit Documentation (#38)
SeasonedSoftware/croods: src/docs/Edit.mdx

---
name: Edit
route: /edit
order: 100
---

# Edit

The Edit component is a union of Info and New.

> This component does a `GET` request to the `path` like Info and its update function, 

Reviewer: ```suggestion
This component firstly does a `GET` request to the `path` like Info. Then its update function, when executed, does a `PUT` request with the JSONified data, like the New component.
```

>   when executed, does a `PUT` request with the jsonified parameter as request body, like New.

Reviewer: ```suggestion
```

>   when executed, does a `PUT` request with the jsonified parameter as request body, like New.

Reviewer: ```suggestion
```

## API

|Property|Type|Required|Default|Description|
|--- |--- |--- |--- |--- |
|id|String/Number|true|-|Ex: 1234 or '1234'|
|name|String|true|-|Defines the deep properties used in the component. Ex: foo.bar|
|render|Func|true|-|A function returning a React Node. Ex: ({ info = {}, update = (id, ...attributes), updating = bool, error }, props) -> < JSX />|
|renderUpdated|Func|false|-|A function returning a React Node. Ex: (updated = {}) -> < JSX />|
|parentId|String/Number|false|-|Parent element's Id, for more specificity in changes.|
|path|String|false|`/${name}/${id}`|Ex: /foo/bar-_12?&=|
|parseInfoResponse|Func|false|-|Parse specific responses to adjust the API to croods patterns. Ex: (json, response, requestAttributes) -> Object|
|parseUpdateResponse|Func|false|-||

## How to use

You have to add a reducer with the same name that you will pass to Edit.

```javascript
import { createReducer } from 'croods'

const reducers = {
  foobar: createReducer('foobar'),
}
```

### render

The render function receives four parameters: **info, update, updating and error**.

- info: the response body of the `GET` request.
- update: the function to make the `PUT` request.

> - updating: the bool that identifies when the `PUT` request is being done and it is not finished.

Reviewer: ```suggestion
- updating: a boolean that identifies when the `PUT` request is being done and it is not finished.
```

> - error: The error message in case that update fails.

Reviewer: ```suggestion
- error: The error message in case the update fails.
```

### renderUpdated

> In order to do something after update, use renderUpdated function to

Reviewer: ```suggestion
In order to do something after the `update`call, use renderUpdated function to
```

>  redirect to other page or show the resource updated/some message/anything in the current page.

Reviewer: ```suggestion
 redirect to other page or show the updated resource, some message or any visual feedback in the current page.
```

 This function receives the response body as parameter.

## Example

> You can see a working example using this component in [here](https://github.com/SeasonedSoftware/croods/blob/master/example/src/colors/Edit.js).

Reviewer: ```suggestion
You can see a working sample of this component's usage [here](https://github.com/SeasonedSoftware/croods/blob/master/example/src/colors/Edit.js).
```
